### PR TITLE
Update manuskript to 0.12, add desktop file

### PR DESCRIPTION
**Pull request by dcbaker**
###### Motivation for this change

- Update to latest version
- Add desktop file for Linux desktops
- 
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/editors/manuskript/default.nix [hunk 1]
```diff
@@ -1,62 +1,73 @@
...
-    plan everything before writing.  The snowflake method can help you
-    grow your idea into a book, by leading you step by step and asking
-    you questions to go deeper. While writing, keep track of notes
-    about every characters, plot, event, place in your story.
-
-    Develop complex characters and keep track of all useful infos.
-    Create intricate plots, linked to your characters, and use them to
-    outline your story. Organize your ideas about the world your
-    characters live in.
+    installPhase = ''
```
Comment (steveeJ) [inline pkgs/applications/editors/manuskript/default.nix:41]: i'd put the `let desktopItem ... in ...` here directly, which would also avoid reformatting the whole file.

Comment (dcbaker) [reply pkgs/applications/editors/manuskript/default.nix]: @steveeJ: okay, done.

File: pkgs/applications/editors/manuskript/default.nix [hunk 2]
```diff
@@ -28,11 +28,22 @@ python3Packages.buildPythonApplication rec {
...
+        exec = "manuskript";
+        desktopName = "Manuskript";
+        genericName = "Manuskript";
+        categories = "Office;";
+      };
+    in
+      ''
+        mkdir -p $out/share/${pname}
+        cp -av  bin/ i18n/ libs/ manuskript/ resources/ icons/ $out
+        cp -a ${desktopItem}/share/applications $out/share
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/manuskript/default.nix:44]: Please use copyDesktopItems instead of copying manually.

Comment (dcbaker) [reply pkgs/applications/editors/manuskript/default.nix:44]: I tried that, but it doesn't seem to work by just putting it on the nativeBuildInputs when a manual `installPhase` is set, nor could I find any documentation or examples of other derivations doing that to copy. Is there some documentation I'm missing?

**Comment by c0bw3b** [comment]
Package update done in #153231

File: pkgs/applications/editors/manuskript/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/applications/editors/manuskript/default.nix:44]: You need to add runHooks to the custom installPhase https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/audio/bucklespring/default.nix#L40-L48.

**Comment by wegank** [comment]
Update done in #153231.

If the change in the PR is still relevant, please open a new PR with the suggested modifications.

**Pull request closed without merging**